Opserver.Core/Data/SQL/SQLInstance.Databases.cs

```diff
@@ -99,6 +100,152 @@ public partial class SQLInstance
+                    conn.ChangeDatabase(databaseName);
+                    return conn.Query<DatabaseResolveSqlHandleToQueryIds>(GetFetchSQL<DatabaseResolveSqlHandleToQueryIds>(),
+                        new {databaseName = databaseName, sqlHandles = listOfHandles}).AsList();
+                },
+                (TimeSpan?) null ?? 0.Minutes(),
+                staleDuration: 0.Minutes());
+
+            if (!queryIdResults.Successful)
+                return;
+            foreach (var cache in results.Data)
+            {
+                cache.QueryIds = new List<long>();
+                cache.PlanIds = new List<long>();
+                var matching = queryIdResults.Data
+                    .Where(x => x.SqlHandle.SequenceEqual(cache.SqlHandle))
+                    .ToList();
+                if (matching.Any())
+                {
+                    
+                    cache.QueryIds = matching.Select(x => x.QueryId).Distinct().ToList();
+                    cache.PlanIds = matching.Select(x => x.PlanId).Distinct().ToList();
+                }
+            }
+        }
+
+        public class DatabaseResolveSqlHandleToQueryIds : ISQLVersioned
+        {
+            public long QueryId { get; internal set; }
+            public long PlanId { get; internal set; }
+            public byte[] SqlHandle { get; internal set; }
+            public Version MinVersion => SQLServerVersions.SQL2008.RTM;
+            public string GetFetchSQL(Version v) => @"SELECT eqs.query_hash, qsp.query_plan_hash, eqs.last_compile_batch_sql_handle as SqlHandle,
+qsp.query_id as queryId, qsp.plan_id as planId
+FROM sys.query_store_query eqs INNER JOIN sys.query_store_plan qsp
+ON eqs.query_id = qsp.query_id
+WHERE eqs.last_compile_batch_sql_handle in @sqlHandles
+";
+        }
+
+        class RemoveSpacesMap<T> : Dapper.SqlMapper.ITypeMap
+        {
+
+            private readonly Type _type;
+            private readonly List<PropertyInfo> _properties;
+
+            public RemoveSpacesMap()
+            {
+                _type = typeof(T);
+                _properties = GetSettableProps(_type);
+            }
+
+            public SqlMapper.IMemberMap GetMember(string columnName)
+            {
+                columnName = columnName.Replace(" ", "");
+                var property = _properties.Find(p => string.Equals(p.Name, columnName, StringComparison.Ordinal))
+                               ?? _properties.Find(p => string.Equals(p.Name, columnName, StringComparison.OrdinalIgnoreCase));
+
+                if (property == null )
+                {
+                    property = _properties.Find(p => string.Equals(p.Name, columnName.Replace("_", ""), StringComparison.Ordinal))
+                               ?? _properties.Find(p => string.Equals(p.Name, columnName.Replace("_", ""), StringComparison.OrdinalIgnoreCase));
+                }
+
+                if (property != null)
+                    return new SimpleMemberMap(columnName, property);
```

Comment:
Does dapper cache this mappings once resolved?

Reply:
I suspect it doesn't...